src/main/java/io/nats/client/NatsThread.java

```diff
@@ -33,7 +29,7 @@ public NatsThread(Runnable runnable, String name) {
         super(runnable, name + "-" + created.incrementAndGet());
         setUncaughtExceptionHandler(new Thread.UncaughtExceptionHandler() {
             public void uncaughtException(Thread t, Throwable e) {
-                logger.debug("UNCAUGHT in thread {}", t.getName(), e);
+               e.printStackTrace();
```

Comment:
This is kinda unrelated, but one issue we had with the Java NATS client at WK was how it started a thread per subscription, which was super resource-heavy for some use cases we had. Would be nice if we could get away from managing threads ourself as much as possible and make subs more lightweight.